bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1331 @@
+                                }
+                                break;
+                            case CHANNEL_SOUNDPROGRAM:
+                                if (zone.equals(zoneToUpdate)) {
+                                    updateState(channelUID, StringType.valueOf(soundProgramState));
+                                }
+                                break;
+                            case CHANNEL_SLEEP:
+                                if (zone.equals(zoneToUpdate)) {
+                                    updateState(channelUID, new DecimalType(sleepState));
+                                }
+                                break;
+                        } // END switch (channelWithoutGroup)
+                    } // END IsLinked
+                }
+                break;
+            case "999":
+                logger.info("YXC - Nothing to do! - {} ({})", thingLabel, zoneToUpdate);
+                break;
+        }
+    }
+
+    private void updatePresets(int value) {
+        String inputText = "";
+        tmpString = getPresetInfo(); // Without zone
+        int presetCounter = 0;
+        int currentPreset = 0;
+        try {
+            JsonElement jsonTree = parser.parse(tmpString);
+            JsonObject jsonObject = jsonTree.getAsJsonObject();
+            JsonArray presetsArray = jsonObject.getAsJsonArray("preset_info");
+            List<StateOption> optionsPresets = new ArrayList<>();
+            inputText = getLastInput(); // Without zone
+            for (JsonElement pr : presetsArray) {
+                presetCounter = presetCounter + 1;
+                JsonObject presetObject = pr.getAsJsonObject();
+                String text = presetObject.get("text").getAsString();
+                if (!text.equals("")) {
+                    optionsPresets.add(new StateOption(String.valueOf(presetCounter),
+                            "#" + String.valueOf(presetCounter) + " " + text));
+                    if (inputText.equals(text)) {
+                        currentPreset = presetCounter;
+                    }
+                }
+            }
+            if (value != 0) {
+                currentPreset = value;
+            }
+            for (Channel channel : getThing().getChannels()) {
+                ChannelUID channelUID = channel.getUID();
+                channelWithoutGroup = channelUID.getIdWithoutGroup();
+                if (isLinked(channelUID)) {
+                    switch (channelWithoutGroup) {
+                        case CHANNEL_SELECTPRESET:
+                            stateDescriptionProvider.setStateOptions(channelUID, optionsPresets);
+                            updateState(channelUID, StringType.valueOf(String.valueOf(currentPreset)));
+                            break;
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.info("Something went wrong with fetching Presets");
```

Comment:
Better log the exception's message. Please check all.

Reply:
These have been revised, removed try/catch block as checks are done upfront
Will be available with next push